python/lsst/pipe/tasks/measurePsf.py

```diff
@@ -278,12 +277,16 @@ def run(self, exposure, sources, expId=0, matches=None):
         reserveList = []
 
         if self.config.reserveFraction > 0:
-            random.seed(self.config.reserveSeed*expId)
-            reserveList = random.sample(psfCandidateList,
-                                        int((self.config.reserveFraction)*len(psfCandidateList)))
-
-            for cand in reserveList:
-                psfCandidateList.remove(cand)
+            # Note that the seed can't be set to 0, so guard against an improper expId.
+            random = afwMath.Random(seed=self.config.reserveSeed*(expId if expId else 1))
+            reserveList = []
+            n = len(psfCandidateList)
+            for i in range(int(n * self.config.reserveFraction)):
```

Comment:
https://developer.lsst.io/coding/python_style_guide.html?highlight=whitespace#binary-operators-should-be-surrounded-by-a-single-space-except-for

Reply:
Still...